flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/OptionSerializer.scala

```diff
@@ -65,7 +65,7 @@ class OptionSerializer[A](val elemSerializer: TypeSerializer[A])
     case Some(a) =>
       target.writeBoolean(true)
       elemSerializer.serialize(a, target)
-    case None =>
+    case _ =>
```

Comment:
This would represent `null` the same way as `None`. However, `null` is not the same as `None` so this is incorrect. 

`writeBoolean` serializes a whole byte: `0` for `false` and `1` for `true`. 
We should change `writeBoolean` to `writeByte` and keep the current mapping and add `2` for `null`.